server/images.go

```diff
@@ -143,62 +143,59 @@ func (m *Model) PostResponseTemplate(p PromptVars) (string, error) {
 	return Prompt(post, p)
 }
 
-func (m *Model) ChatPrompt(msgs []api.Message) (string, []api.ImageData, error) {
+type ChatHistory struct {
+	Prompts       []PromptVars
+	CurrentImages []api.ImageData
+	LastSystem    string
```

Comment:
The change here (other than the struct) is returning the `LastSystem` which is used to keep track of what the most recent system message is later in the code.